crates/red_knot_test/src/lib.rs

```diff
@@ -96,7 +97,15 @@ fn run_test(db: &mut db::Db, test: &parser::MarkdownTest) -> Result<(), Failures
                 parsed.errors()
             );
 
-            match matcher::match_file(db, test_file.file, check_types(db, test_file.file)) {
+            let diagnostics: Vec<_> =
+                // The accumulator returns all diagnostics from all files. We're only interested into
+                // diagnostics from this file.
```

Comment:
Should we be doing a workspace check once and then splitting the diagnostics by file, instead of re-doing the check for each file and filtering the diagnostics?

Doesn't need to be done in this PR, regardless. And maybe tests are small enough that it really doesn't matter.

Reply:
The method to check all files is defined in `red_knot_workspace` on which this crate doesn't depend. It also requires creating a workspace, which is currently somewhat awkward and it does concurrent checking, which we don't want. That's why I think the current approach is still the right approach for testing